apps/files/tasks.py

```diff
@@ -31,8 +38,131 @@ def extract_file(viewer, **kw):
                      (viewer, err))
         else:
             msg.save(_('There was an error accessing file %s.') % viewer)
-        task_log.error('[1@%s] Error unzipping: %s' %
-                       (extract_file.rate_limit, err))
+        log.error('[1@%s] Error unzipping: %s' %
+                  (extract_file.rate_limit, err))
     finally:
         # Always delete the flag so the file never gets into a bad state.
         flag.delete()
+
+
+def fix_let_scope_bustage(*files):
+    """Needed for bug 1224444: rewrite all the top level `let` to `var`.
+
+    Usage:
+    >>> fix_let_scope_bustage(file1, file2, file3, ...)
+    ('', '')  # First one is stdout, second is stderr.
+    """
+    fixage_script = os.path.join('scripts', 'rewrite.js')
+    # We need to force the version to 180, because in 185 there were some
+    # breaking changes: using `function` for generators was deprecated, but a
+    # lot of the current add-ons are using that.
+    process = spidermonkey.Spidermonkey(code='version(180)',
+                                        script_file=fixage_script,
+                                        script_args=files)
+    return process.communicate()
+
+
+def fix_let_scope_bustage_in_xpi(xpi_path):
+    """Rewrite all the top level `let` to `var` in an XPI."""
+    files_to_fix = []
+    with repack(xpi_path) as folder:
+        for root, dirs, files in os.walk(folder):
+            for file_ in files:
+                if file_.endswith('.js'):
+                    # We only want to fix javascript files.
+                    files_to_fix.append(os.path.join(root, file_))
+        fix_let_scope_bustage(*files_to_fix)
+
+
+MAIL_SUBJECT = u'Mozilla Add-ons: {addon} has been automatically fixed on AMO'
+MAIL_MESSAGE = u"""
+Your add-on, {addon}, has been automatically fixed for future versions of
+Firefox (see
+https://blog.mozilla.org/addons/2015/10/14/breaking-changes-let-const-firefox-nightly-44/).
+The fixing process involved repackaging the add-on files and adding the string
+'.1-let-fixed' to their versions numbers. We only fixed the files for the
+last uploaded version.
```

Comment:
I don't think the bustage is that urgent, so it *should* be sufficient to simply sign the latest submitted version. If it's pending review, it should be reviewed before the bustage happens, right?

I'd rather keep this as simple and contained as possible, it's already quite a mess :/